js/packages/mobile-wallet-adapter-protocol/src/transact.ts

```diff
@@ -171,7 +173,18 @@ export async function transact<TReturn>(
                         state.associationPublicKey,
                         state.ecdhPrivateKey,
                     );
-                    state = { __type: 'connected', sharedSecret };
+                    const sessionPropertiesBuffer = responseBuffer.slice(ENCODED_PUBLIC_KEY_LENGTH_BYTES);
+                    const sessionProperties = sessionPropertiesBuffer.byteLength !== 0 
+                        ? await (async () => {
+                            const sequenceNumberVector = sessionPropertiesBuffer.slice(0, SEQUENCE_NUMBER_BYTES);
```

Comment:
Testing my understanding:

There's now a `sequenceNumber` also contained within the `sessionProperties` buffer

Reply:
the session properties are encrypted, in the same way we encrypt ever other payload that is sent over the websocket following the key exchange. 

more details about the encrypted message wrapping in the spec [here](https://solana-mobile.github.io/mobile-wallet-adapter/spec/spec.html#encrypted-message-wrapping).